commit b07b763958e5483cd1a8247283bcb7dfdd9f9c12
Author:     Yuhong Wen <yuhongw@nvidia.com>
AuthorDate: Tue Aug 29 11:09:15 2023 -0400
Commit:     GitHub <noreply@github.com>
CommitDate: Tue Aug 29 11:09:15 2023 -0400

    Remove fobs calls (#1960)
    
    * Removed the extra fobs.dumps() calls.
    
    * removed more fobs.dump().
    
    * Removed more fobs.dumps().
    
    * Removed additional Fobs.dumps().
    
    * Removed more Fobs.dumps() calls.
    
    * Removed no use import.
---
 nvflare/private/fed/client/client_executor.py | 17 +++++++--------
 nvflare/private/fed/client/scheduler_cmds.py  | 11 +++++-----
 nvflare/private/fed/server/server_engine.py   | 31 ++++++++-------------------
 3 files changed, 22 insertions(+), 37 deletions(-)

diff --git a/nvflare/private/fed/client/client_executor.py b/nvflare/private/fed/client/client_executor.py
index 5afdad37..c4b4f364 100644
--- a/nvflare/private/fed/client/client_executor.py
+++ b/nvflare/private/fed/client/client_executor.py
@@ -26,7 +26,6 @@ from nvflare.apis.resource_manager_spec import ResourceManagerSpec
 from nvflare.fuel.common.exit_codes import PROCESS_EXIT_REASON, ProcessExitCode
 from nvflare.fuel.f3.cellnet.core_cell import FQCN
 from nvflare.fuel.f3.cellnet.defs import MessageHeaderKey, ReturnCode
-from nvflare.fuel.utils import fobs
 from nvflare.private.defs import CellChannel, CellChannelTopic, JobFailureMsgKey, new_cell_message
 from nvflare.security.logging import secure_format_exception, secure_log_traceback
 
@@ -222,7 +221,7 @@ class ProcessExecutor(ClientExecutor):
             with self.lock:
                 data = {}
                 fqcn = FQCN.join([self.client.client_name, job_id])
-                request = new_cell_message({}, fobs.dumps(data))
+                request = new_cell_message({}, data)
                 return_data = self.client.cell.send_request(
                     target=fqcn,
                     channel=CellChannel.CLIENT_COMMAND,
@@ -232,7 +231,7 @@ class ProcessExecutor(ClientExecutor):
                 )
                 return_code = return_data.get_header(MessageHeaderKey.RETURN_CODE)
                 if return_code == ReturnCode.OK:
-                    status_message = fobs.loads(return_data.payload)
+                    status_message = return_data.payload
                     self.logger.debug("check status from process listener......")
                     return status_message
                 else:
@@ -256,7 +255,7 @@ class ProcessExecutor(ClientExecutor):
             with self.lock:
                 data = {}
                 fqcn = FQCN.join([self.client.client_name, job_id])
-                request = new_cell_message({}, fobs.dumps(data))
+                request = new_cell_message({}, data)
                 return_data = self.client.cell.send_request(
                     target=fqcn,
                     channel=CellChannel.CLIENT_COMMAND,
@@ -266,7 +265,7 @@ class ProcessExecutor(ClientExecutor):
                 )
                 return_code = return_data.get_header(MessageHeaderKey.RETURN_CODE)
                 if return_code == ReturnCode.OK:
-                    run_info = fobs.loads(return_data.payload)
+                    run_info = return_data.payload
                     return run_info
                 else:
                     return {}
@@ -288,7 +287,7 @@ class ProcessExecutor(ClientExecutor):
             with self.lock:
                 data = {"command": AdminCommandNames.SHOW_ERRORS, "data": {}}
                 fqcn = FQCN.join([self.client.client_name, job_id])
-                request = new_cell_message({}, fobs.dumps(data))
+                request = new_cell_message({}, data)
                 return_data = self.client.cell.send_request(
                     target=fqcn,
                     channel=CellChannel.CLIENT_COMMAND,
@@ -317,7 +316,7 @@ class ProcessExecutor(ClientExecutor):
             with self.lock:
                 data = {"command": AdminCommandNames.RESET_ERRORS, "data": {}}
                 fqcn = FQCN.join([self.client.client_name, job_id])
-                request = new_cell_message({}, fobs.dumps(data))
+                request = new_cell_message({}, data)
                 self.client.cell.fire_and_forget(
                     targets=fqcn,
                     channel=CellChannel.CLIENT_COMMAND,
@@ -347,7 +346,7 @@ class ProcessExecutor(ClientExecutor):
                         child_process = self.run_processes[job_id][RunProcessKey.CHILD_PROCESS]
                         data = {}
                         fqcn = FQCN.join([self.client.client_name, job_id])
-                        request = new_cell_message({}, fobs.dumps(data))
+                        request = new_cell_message({}, data)
                         self.client.cell.fire_and_forget(
                             targets=fqcn,
                             channel=CellChannel.CLIENT_COMMAND,
@@ -414,7 +413,7 @@ class ProcessExecutor(ClientExecutor):
             if process_status == ClientStatus.STARTED:
                 data = {"command": AdminCommandNames.ABORT_TASK, "data": {}}
                 fqcn = FQCN.join([self.client.client_name, job_id])
-                request = new_cell_message({}, fobs.dumps(data))
+                request = new_cell_message({}, data)
                 self.client.cell.fire_and_forget(
                     targets=fqcn,
                     channel=CellChannel.CLIENT_COMMAND,
diff --git a/nvflare/private/fed/client/scheduler_cmds.py b/nvflare/private/fed/client/scheduler_cmds.py
index 3a1e7c2e..58828121 100644
--- a/nvflare/private/fed/client/scheduler_cmds.py
+++ b/nvflare/private/fed/client/scheduler_cmds.py
@@ -19,7 +19,6 @@ from nvflare.apis.event_type import EventType
 from nvflare.apis.fl_constant import FLContextKey, ReturnCode, SystemComponents
 from nvflare.apis.resource_manager_spec import ResourceConsumerSpec, ResourceManagerSpec
 from nvflare.apis.shareable import Shareable
-from nvflare.fuel.utils import fobs
 from nvflare.private.admin_defs import Message
 from nvflare.private.defs import ERROR_MSG_PREFIX, RequestHeader, SysCommandTopic, TrainingTopic
 from nvflare.private.fed.client.admin import RequestProcessor
@@ -65,7 +64,7 @@ class CheckResourceProcessor(RequestProcessor):
         with engine.new_context() as fl_ctx:
             try:
                 job_id = req.get_header(RequestHeader.JOB_ID, "")
-                resource_spec = fobs.loads(req.body)
+                resource_spec = req.body
                 fl_ctx.set_prop(key=FLContextKey.CLIENT_RESOURCE_SPECS, value=resource_spec, private=True, sticky=False)
 
                 fl_ctx.set_prop(FLContextKey.CURRENT_JOB_ID, job_id, private=True, sticky=False)
@@ -85,7 +84,7 @@ class CheckResourceProcessor(RequestProcessor):
         result.set_header(ShareableHeader.IS_RESOURCE_ENOUGH, is_resource_enough)
         result.set_header(ShareableHeader.RESOURCE_RESERVE_TOKEN, token)
 
-        return Message(topic="reply_" + req.topic, body=fobs.dumps(result))
+        return Message(topic="reply_" + req.topic, body=result)
 
 
 class StartJobProcessor(RequestProcessor):
@@ -98,7 +97,7 @@ class StartJobProcessor(RequestProcessor):
 
         allocated_resources = None
         try:
-            resource_spec = fobs.loads(req.body)
+            resource_spec = req.body
             job_id = req.get_header(RequestHeader.JOB_ID)
             token = req.get_header(ShareableHeader.RESOURCE_RESERVE_TOKEN)
         except Exception as e:
@@ -141,13 +140,13 @@ class CancelResourceProcessor(RequestProcessor):
 
         with engine.new_context() as fl_ctx:
             try:
-                resource_spec = fobs.loads(req.body)
+                resource_spec = req.body
                 token = req.get_header(ShareableHeader.RESOURCE_RESERVE_TOKEN)
                 resource_manager.cancel_resources(resource_requirement=resource_spec, token=token, fl_ctx=fl_ctx)
             except Exception:
                 result.set_return_code(ReturnCode.EXECUTION_EXCEPTION)
 
-        return Message(topic="reply_" + req.topic, body=fobs.dumps(result))
+        return Message(topic="reply_" + req.topic, body=result)
 
 
 class ReportResourcesProcessor(RequestProcessor):
diff --git a/nvflare/private/fed/server/server_engine.py b/nvflare/private/fed/server/server_engine.py
index d0a5ea39..cb56834d 100644
--- a/nvflare/private/fed/server/server_engine.py
+++ b/nvflare/private/fed/server/server_engine.py
@@ -49,7 +49,6 @@ from nvflare.apis.workspace import Workspace
 from nvflare.fuel.f3.cellnet.core_cell import FQCN, CoreCell
 from nvflare.fuel.f3.cellnet.defs import MessageHeaderKey
 from nvflare.fuel.f3.cellnet.defs import ReturnCode as CellMsgReturnCode
-from nvflare.fuel.utils import fobs
 from nvflare.fuel.utils.argument_utils import parse_vars
 from nvflare.fuel.utils.network_utils import get_open_ports
 from nvflare.fuel.utils.zip_utils import zip_directory_to_bytes
@@ -72,18 +71,6 @@ from .server_engine_internal_spec import EngineInfo, ServerEngineInternalSpec
 from .server_status import ServerStatus
 
 
-class ClientConnection:
-    def __init__(self, client):
-        self.client = client
-
-    def send(self, data):
-        data = fobs.dumps(data)
-        self.client.send(data)
-
-    def recv(self):
-        return self.client.recv()
-
-
 class ServerEngine(ServerEngineInternalSpec):
     def __init__(self, server, args, client_manager: ClientManager, snapshot_persistor, workers=3):
         """Server engine.
@@ -551,7 +538,7 @@ class ServerEngine(ServerEngineInternalSpec):
         return self.client_manager.clients
 
     def _retrieve_clients_data(self, job_id):
-        request = new_cell_message({CellMessageHeaderKeys.JOB_ID: job_id}, fobs.dumps({}))
+        request = new_cell_message({CellMessageHeaderKeys.JOB_ID: job_id}, {})
         return_data = self.server.cell.send_request(
             target=FQCN.ROOT_SERVER,
             channel=CellChannel.SERVER_PARENT_LISTENER,
@@ -565,7 +552,7 @@ class ServerEngine(ServerEngineInternalSpec):
             self.logger.debug(f"cannot retrieve clients from parent: {rc}")
             return None
 
-        data = fobs.loads(return_data.payload)
+        data = return_data.payload
         clients = data.get(ServerCommandKey.CLIENTS, None)
         if clients is None:
             self.logger.error(f"parent failed to return clients info for job {job_id}")
@@ -576,7 +563,7 @@ class ServerEngine(ServerEngineInternalSpec):
             execution_error = fl_ctx.get_prop(FLContextKey.FATAL_SYSTEM_ERROR, False)
             data = {"execution_error": execution_error}
             job_id = fl_ctx.get_job_id()
-            request = new_cell_message({CellMessageHeaderKeys.JOB_ID: job_id}, fobs.dumps(data))
+            request = new_cell_message({CellMessageHeaderKeys.JOB_ID: job_id}, data)
             return_data = self.server.cell.fire_and_forget(
                 targets=FQCN.ROOT_SERVER,
                 channel=CellChannel.SERVER_PARENT_LISTENER,
@@ -589,7 +576,7 @@ class ServerEngine(ServerEngineInternalSpec):
     ):
         with self.lock:
             fqcn = FQCN.join([FQCN.ROOT_SERVER, job_id])
-            request = new_cell_message({}, fobs.dumps(command_data))
+            request = new_cell_message({}, command_data)
             if timeout <= 0.0:
                 self.server.cell.fire_and_forget(
                     targets=fqcn,
@@ -610,7 +597,7 @@ class ServerEngine(ServerEngineInternalSpec):
             )
             rc = return_data.get_header(MessageHeaderKey.RETURN_CODE, CellMsgReturnCode.OK)
             if rc == CellMsgReturnCode.OK:
-                result = fobs.loads(return_data.payload)
+                result = return_data.payload
             else:
                 result = None
         return result
@@ -726,7 +713,7 @@ class ServerEngine(ServerEngineInternalSpec):
             # assume server resource is unlimited
             if site_name == "server":
                 continue
-            request = Message(topic=TrainingTopic.CHECK_RESOURCE, body=fobs.dumps(resource_requirements))
+            request = Message(topic=TrainingTopic.CHECK_RESOURCE, body=resource_requirements)
             request.set_header(RequestHeader.JOB_ID, job_id)
             client = self.get_client_from_name(site_name)
             if client:
@@ -743,7 +730,7 @@ class ServerEngine(ServerEngineInternalSpec):
                     self.logger.error(f"Client reply error: {r.reply.body}")
                     result[site_name] = (False, "")
                 else:
-                    resp = fobs.loads(r.reply.body)
+                    resp = r.reply.body
                     result[site_name] = (
                         resp.get_header(ShareableHeader.IS_RESOURCE_ENOUGH, False),
                         resp.get_header(ShareableHeader.RESOURCE_RESERVE_TOKEN, ""),
@@ -760,7 +747,7 @@ class ServerEngine(ServerEngineInternalSpec):
             is_resource_enough, token = result
             if is_resource_enough and token:
                 resource_requirements = resource_reqs[site_name]
-                request = Message(topic=TrainingTopic.CANCEL_RESOURCE, body=fobs.dumps(resource_requirements))
+                request = Message(topic=TrainingTopic.CANCEL_RESOURCE, body=resource_requirements)
                 request.set_header(ShareableHeader.RESOURCE_RESERVE_TOKEN, token)
                 client = self.get_client_from_name(site_name)
                 if client:
@@ -773,7 +760,7 @@ class ServerEngine(ServerEngineInternalSpec):
         for site, dispatch_info in client_sites.items():
             resource_requirement = dispatch_info.resource_requirements
             token = dispatch_info.token
-            request = Message(topic=TrainingTopic.START_JOB, body=fobs.dumps(resource_requirement))
+            request = Message(topic=TrainingTopic.START_JOB, body=resource_requirement)
             request.set_header(RequestHeader.JOB_ID, job_id)
             request.set_header(ShareableHeader.RESOURCE_RESERVE_TOKEN, token)
             client = self.get_client_from_name(site)
